packages/contracts/test/core/dao/dao.ts

```diff
@@ -1198,90 +1225,175 @@ describe('DAO', function () {
+
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature))
+        .to.be.revertedWithCustomError(dao, 'Unauthorized')
+        .withArgs();
+
+      // Grant the permission to validate signatures to the caller unconditionally
+      await dao.grant(
+        dao.address,
+        caller.address,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID
+      );
+
+      // The caller can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
+
+      // Because the caller is allowed unconditionally, the signature is always valid.
+      expect(
+        await dao.connect(caller).isValidSignature(hash, goodSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+      expect(
+        await dao.connect(caller).isValidSignature(hash, badSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
     });
 
-    it('should return the validators response', async () => {
-      const ERC1271MockFactory = new ERC1271Mock__factory(signers[0]);
-      const erc1271Mock = await ERC1271MockFactory.deploy();
+    it('allows generic signature validation', async () => {
+      const caller = signers[0];
+      const signer = signers[1];
+      const other = signers[3];
+
+      const message = 'Hi!';
+      const hash = ethers.utils.hashMessage(message);
+      const goodSignature = await signer.signMessage(message);
+      const badSignature = await signer.signMessage('Bye!');
+      expect(badSignature).to.not.equal(goodSignature);
 
-      await dao.setSignatureValidator(erc1271Mock.address);
+      // Try to call with caller but caller has no permission
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature))
+        .to.be.revertedWithCustomError(dao, 'Unauthorized')
+        .withArgs();
+
+      const mockCondition = await new PermissionConditionMock__factory(
+        caller
+      ).deploy();
+
+      // Grant the permission to validate signatures to the caller conditionally (granting it unconditionally is not possible in combination with `_who: ANY_ADDR`)
+      await dao.grantWithCondition(
+        dao.address,
+        ANY_ADDR,
+        PERMISSION_IDS.VALIDATE_SIGNATURE_PERMISSION_ID,
+        mockCondition.address
+      );
+
+      // Any contract can validate signatures now.
+      expect(await dao.connect(caller).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
+      expect(await dao.connect(other).isValidSignature(hash, goodSignature)).to
+        .not.be.reverted;
+
+      // Because we use a mock condition always returning true, the signature is always valid.
       expect(
-        await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00')
-      ).to.be.eq('0x41424344');
+        await dao.connect(caller).isValidSignature(hash, goodSignature)
+      ).to.equal(VALID_ERC1271_SIGNATURE);
+      expect(
```

Comment:
Could you switch the mock to return false? This way I would like to test that the test actually sets the mocked condition. Because if the mocked condition is not set in this test, the test would still pass and it would be another failsafe

Reply:
Changed in 25b7704e70b50f8b8a546560083e551d0aeab194